week-1-ts-ecosystem/w01e01-typescript-basics/src/models/user/user.model.ts

```diff
@@ -0,0 +1,18 @@
+import {ADULT_AGE} from "../../core";
+import {hasGivenAge} from "../../utils";
+import {Addres} from "../address";
+
+export interface User {
+    name: string;
+    age: number;
+    email: string;
+    address: Addres;
+}
+
+export namespace User {
+    export function hasAddress(user: User): boolean {
+        return Boolean(user.address)
+    }
+
+    export const isAdult  = hasGivenAge(ADULT_AGE);
+}
```

Comment:
Tutaj rozumiem pomysł spakowania tego do `namespace` jednak - skąd brak konsekwencji z i wrzucenie do `age.utils`
hasGivenAge ?

2 sprawa to nazwa samego namespace, `User` będzie sugerowało użycie tych elementów tak jakby `User` był klasą a same elementy - statycznymi metodami i polami.

w mojej opinii da to złudne wrażenie że `User` to klasa a nie "rozpływający się" po buildzie: interface

Reply:
z hasGivenAge to mój błąd, próbowałem różnych opcji i nie przeczytałem kodu ponownie przed oddaniem do pr : ]

Takie namespace'y ściśle związane z modelem używamy u nas w projekcie i moim zdaniem całkiem spoko się to sprawdza, ot statyczne metody bez klas ale musi być konsekwentne w całym projekcie, klasy robimy tylko dla angularowych klocków.  Ofc mogę zmienić, wiem że to 'kontrowersyjny' pattern

Reply:
Jak dla mnie Twoje wyjaśnienie @zajewskp - broni pattern. Jak się sprawdza + jest konsekwencja - jestem za.